### ISSUE TITLE
Support for node 0.12

### ISSUE THREAD

**Issue by marlonmantilla** [Sembiance/mhash#16]
Hi guys I'm having issues when installing mash in Node 0.12.0 

``` 
../mhash.cc:129:2: error: use of undeclared identifier 'NanScope'
        NanScope();
        ^
../mhash.cc:134:26: error: use of undeclared identifier 'args'; did you mean 'uv_process_options_s::args'?
        String::Utf8Value       name(args[0]->ToString());
                                     ^~~~
                                     uv_process_options_s::args
/Users/marman/.node-gyp/0.12.0/deps/uv/include/uv.h:822:10: note: 'uv_process_options_s::args' declared here
  char** args;
         ^
../mhash.cc:134:26: error: invalid use of non-static data member 'args'
        String::Utf8Value       name(args[0]->ToString());
                                     ^~~~
../mhash.cc:137:18: error: use of undeclared identifier 'NanNull'
                NanReturnValue(NanNull());
                               ^
../mhash.cc:139:5: error: use of undeclared identifier 'args'; did you mean 'uv_process_options_s::args'?
        if(args[1]->IsString())
           ^~~~
           uv_process_options_s::args
/Users/marman/.node-gyp/0.12.0/deps/uv/include/uv.h:822:10: note: 'uv_process_options_s::args' declared here
  char** args;
         ^
../mhash.cc:139:5: error: invalid use of non-static data member 'args'
        if(args[1]->IsString())
           ^~~~
../mhash.cc:141:27: error: use of undeclared identifier 'args'; did you mean 'uv_process_options_s::args'?
                String::Utf8Value       data(args[1]->ToString());
                                             ^~~~
                                             uv_process_options_s::args
/Users/marman/.node-gyp/0.12.0/deps/uv/include/uv.h:822:10: note: 'uv_process_options_s::args' declared here
  char** args;
         ^
../mhash.cc:141:27: error: invalid use of non-static data member 'args'
                String::Utf8Value       data(args[1]->ToString());
                                             ^~~~
../mhash.cc:144:36: error: use of undeclared identifier 'args'; did you mean 'uv_process_options_s::args'?
        else if(node::Buffer::HasInstance(args[1]))
                                          ^~~~
                                          uv_process_options_s::args
/Users/marman/.node-gyp/0.12.0/deps/uv/include/uv.h:822:10: note: 'uv_process_options_s::args' declared here
  char** args;
         ^
../mhash.cc:144:36: error: invalid use of non-static data member 'args'
        else if(node::Buffer::HasInstance(args[1]))
                                          ^~~~
../mhash.cc:146:25: error: use of undeclared identifier 'args'; did you mean 'uv_process_options_s::args'?
                Handle<Object> data = args[1]->ToObject();
                                      ^~~~
                                      uv_process_options_s::args
/Users/marman/.node-gyp/0.12.0/deps/uv/include/uv.h:822:10: note: 'uv_process_options_s::args' declared here
  char** args;
         ^
../mhash.cc:146:25: error: invalid use of non-static data member 'args'
                Handle<Object> data = args[1]->ToObject();
                                      ^~~~
../mhash.cc:151:18: error: use of undeclared identifier 'NanNull'
                NanReturnValue(NanNull());
                               ^
../mhash.cc:153:8: error: use of undeclared identifier 'NanNew'
        ret = NanNew<String>(hashed);
              ^
../mhash.cc:153:15: error: 'String' does not refer to a value
        ret = NanNew<String>(hashed);
                     ^
/Users/marman/.node-gyp/0.12.0/deps/v8/include/v8.h:1599:17: note: declared here
class V8_EXPORT String : public Primitive {
                ^
../mhash.cc:156:2: error: use of undeclared identifier 'NanReturnValue'
        NanReturnValue(ret);
        ^
../mhash.cc:161:15: error: use of undeclared identifier 'NanNew'
        exports->Set(NanNew("hash"), NanNew<FunctionTemplate>(Hash)->GetFunction());
                     ^
../mhash.cc:161:31: error: use of undeclared identifier 'NanNew'
        exports->Set(NanNew("hash"), NanNew<FunctionTemplate>(Hash)->GetFunction());
                                     ^
../mhash.cc:161:38: error: 'FunctionTemplate' does not refer to a value
        exports->Set(NanNew("hash"), NanNew<FunctionTemplate>(Hash)->GetFunction());
                                            ^
/Users/marman/.node-gyp/0.12.0/deps/v8/include/v8.h:3452:17: note: declared here
class V8_EXPORT FunctionTemplate : public Template {
                ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make: *** [Release/obj.target/mhash/mhash.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/marman/.nvm/versions/node/v0.12.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:267:23)
gyp ERR! stack     at ChildProcess.emit (events.js:110:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1067:12)
gyp ERR! System Darwin 14.4.0
gyp ERR! command "node" "/Users/marman/.nvm/versions/node/v0.12.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/marman/Projects/selfie-api/node_modules/mhash
gyp ERR! node -v v0.12.0
gyp ERR! node-gyp -v v1.0.2
gyp ERR! not ok 
npm ERR! Darwin 14.4.0
npm ERR! argv "/Users/marman/.nvm/versions/node/v0.12.0/bin/node" "/Users/marman/.nvm/versions/node/v0.12.0/bin/npm" "install" "mhash" "--python=python2" "--save"
npm ERR! node v0.12.0
npm ERR! npm  v2.5.1
npm ERR! code ELIFECYCLE

npm ERR! mhash@2.0.0 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the mhash@2.0.0 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the mhash package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls mhash
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/marman/Projects/selfie-api/npm-debug.log

```

Node v0.12.0
NPM 2.5.1

Any idea ?

**Comment by lporras** [comment]
hey guys, 

I'm having the same problem with Node 0.10.35 and OSX Yosemite 10.10.4, but version 1.1.1 of mhash installs without problems.

**Comment by Sembiance** [comment]
I was able to reproduce the issues you two described.

I updated a package dependency 'nan' to the latest 1.x version has fixed the problem.

I tested the latest version I just published to npm with both node 0.12.7 and 0.10.35 and both now install it just fine.

Enjoy :)

**Comment by marlonmantilla** [comment]
Yes sir!!! working with Node 0.12.5 thanks @Sembiance
